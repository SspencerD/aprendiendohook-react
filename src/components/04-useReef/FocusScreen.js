import React, { useRef } from 'react'
import '../02-useEffect/effects.css';

//Para mantener una referencia mutable 
export const FocusScreen = () => {

    const inputRef = useRef();
    

    const handleClick =()=>{
        inputRef.current.select();
    }
    return (
        <div>
            <h1>Enfocar Pantalla</h1>
            <hr/>

            <input
            ref={ inputRef }
            className="form-control"
            placeholder="Su Nombre"/>

            <button 
            className="btn btn-outline-primary mt-5"
            onClick={ handleClick }>
                Enfocar
            </button>
            
        </div>
    )
}
