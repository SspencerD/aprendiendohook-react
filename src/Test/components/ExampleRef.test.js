import { shallow } from "enzyme";
import { ExampleRef } from "../../components/04-useReef/ExampleRef";

describe('Prueba en example ref', () => {
    

    const wrapper = shallow(<ExampleRef/>);

test('Debe mostrar correctamente', () => {

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    
});



    test(' debe mostrar el componente <mulipleCustomHooks', () => {

        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

        
    });
});