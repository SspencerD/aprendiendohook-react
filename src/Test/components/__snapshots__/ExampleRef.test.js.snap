// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en example ref Debe mostrar correctamente 1`] = `
<div>
  <h1>
     Ejemplos con useRef 
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Mostrar/Ocultar
  </button>
</div>
`;
